widget/embedded/src/utils/wallets.ts

```diff
@@ -415,98 +359,70 @@ export const getKeplrCompatibleConnectedWallets = (
   );
 };
 
-export function getTokensWithBalance(
-  tokens: TokenWithBalance[],
-  connectedWallets: ConnectedWallet[]
-): TokenWithBalance[] {
-  return tokens.map(({ balance, ...otherProps }) => {
-    const tokenAmount = numberToString(
-      new BigNumber(
-        getBalanceFromWallet(
-          connectedWallets,
-          otherProps.blockchain,
-          otherProps.symbol,
-          otherProps.address
-        )?.amount || ZERO
-      )
-    );
+export function getBalanceFor(
+  token: Token,
+  balances: TokensBalance
+): Balance | null {
+  const balance =
+    balances[`${token.symbol}-${token.blockchain}-${token.address ?? ''}`];
+  return balance ?? null;
+}
 
-    let tokenUsdValue = '';
-    if (otherProps.usdPrice) {
-      tokenUsdValue = numberToString(
-        new BigNumber(
-          getBalanceFromWallet(
-            connectedWallets,
-            otherProps.blockchain,
-            otherProps.symbol,
-            otherProps.address
-          )?.amount || ZERO
-        ).multipliedBy(otherProps.usdPrice)
-      );
-    }
+export function formatBalance(
+  token: Token,
+  balances: TokensBalance
+): Balance | null {
+  const balance = getBalanceFor(token, balances);
+
+  const formattedBalance: Balance | null = balance
+    ? {
+        ...balance,
+        amount: numberToString(
+          balance.amount,
+          BALANCE_MIN_DECIMALS,
+          BALANCE_MAX_DECIMALS
+        ),
+        usdValue: numberToString(
+          balance.usdValue,
+          USD_VALUE_MIN_DECIMALS,
+          USD_VALUE_MAX_DECIMALS
+        ),
+      }
+    : null;
 
-    return {
-      ...otherProps,
-      ...(tokenAmount !== '0' && {
-        balance: { amount: tokenAmount, usdValue: tokenUsdValue },
-      }),
-    };
-  });
+  return formattedBalance;
 }
 
-export function getTokensBalanceFromWalletAndSort(
-  tokens: TokenWithBalance[],
-  connectedWallets: ConnectedWallet[]
-) {
-  const list =
-    connectedWallets.length > 0
-      ? getTokensWithBalance(tokens, connectedWallets)
-      : tokens;
-  list.sort((tokenA, tokenB) => {
-    if (tokenA.balance?.usdValue && tokenB.balance?.usdValue) {
+export function sortTokensByBalance(tokens: Token[], balances: TokensBalance) {
+  tokens.sort((tokenA, tokenB) => {
+    const tokenA_balance = getBalanceFor(tokenA, balances);
+    const tokenB_balance = getBalanceFor(tokenB, balances);
```

Comment:
We could change names to: `token1, token2, token1Balance, token2Balance` to stay with our naming conventions.